spec/captain_hook_spec.rb

```diff
@@ -42,8 +49,17 @@ class ResourceWithHooks
        exclude: [:serve],
        skip_when: ->(_args, kwargs) { !kwargs[:dto] }
 
-  hook :after, methods: [:prepare], hook: PrepareHook.new
-  hook :after, methods: [:invented_one], hook: PrepareHook.new
+  hook :after, methods: %i[prepare invented_one], hook: PrepareHook.new
+
+  hook :around,
+       methods: [:prepare],
+       hook: ManyParametersHook.new,
+       # TODO: Maybe this should be defined in the hook class itself?
```

Comment:
This could be good for next iterations. The idea here would be to delegate to each hook how their parameters are constructed on hook lifecycle